services/src/main/java/org/keycloak/services/resources/admin/IdentityProviderResource.java

```diff
@@ -128,7 +129,17 @@ public Response delete() {
             throw new javax.ws.rs.NotFoundException();
         }
 
-        this.realm.removeIdentityProviderByAlias(this.identityProviderModel.getAlias());
+        String alias = this.identityProviderModel.getAlias();
+        if (this.realm.removeIdentityProviderByAlias(alias)) {
+            Set<IdentityProviderMapperModel> mappers = this.realm.getIdentityProviderMappersByAlias(alias);
+
+            for (IdentityProviderMapperModel mapper : mappers) {
+                if (mapper == null) {
+                    throw new javax.ws.rs.NotFoundException();
```

Comment:
This should be silently ignored since we're deleting anyway.

Reply:
Delete null checking brace